Addresed FIPS compliance issues

xlian-nextgen opened the pull request:
Due to FIPS(Federal information Processing Standards) newer versions of cfgrib encounter an error during runtime. This issue can be addressed by setting the usedforsecurity flag on hashlib md5 to false. Due to the primary use of this project being an API to support a grib engine for xarray this flag can be set to false.

The code change is based on an issue opened previously
https://github.com/ecmwf/cfgrib/issues/383

xlian-nextgen committed:
Addresed FIPS compliance issues (#1)

iainrussell commented:
Hi @xlian-nextgen, many thanks for this contribution - I will merge it, but first can you add a comment with our contributors' agreement text? https://github.com/ecmwf/.github/blob/main/.github/PULL_REQUEST_TEMPLATE.md

barronh commented:
@xlian-nextgen, thanks indeed! With the end-of-life on python3.9, this is starting to pop up in production environments. We have Dockers that fail when they update to newer python.

xlian-nextgen commented:
### Description


### Contributor Declaration

By opening this pull request, I affirm the following:

* All authors agree to the [Contributor License Agreement](https://github.com/ecmwf/codex/blob/main/Legal/contributor_license_agreement.md).
* The code follows the project's coding standards.
* I have performed self-review and added comments where needed.
* I have added or updated tests to verify that my changes are effective and functional.
* I have run all existing tests and confirmed they pass.

xlian-nextgen commented:
Thank you for the review, I have added a comment with the contributors' agreement text. Let me know if you need anything else.

iainrussell committed:
Merge branch 'master' into master

barronh commented:
Thanks for merging this fix. I'm making a note here because I closed the #383 where someone had asked when this might be in a release. At the time of this writing, the pypi version is 0.9.15.1 and does not include this merge. Would it be possible to add this minor feature as a micro to the 0.9.15 series and simply release that? Or will it be part of a larger release sometime soon?

Thanks!
